### PR TITLE
Add option to turn on / off projection

### PR DESCRIPTION
**Pull request by dafeda**
Resolves: #42 

Projection should only be done when the full global state vector has less parameters than N-1 where N are the number or realizations. The projection should not be applied when for example updating parameters one at a time, or a chunk at a time.

### CODE REVIEW

File: src/iterative_ensemble_smoother/experimental.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def ensemble_smoother_update_step_row_scaling(
             D,
             inversion,
             truncation,
+            True,
```
Comment (ManInFez) [inline src/iterative_ensemble_smoother/experimental.py:39]: should we do projection when row_scaling? I thought that was doing update one parameter at a time?

File: src/iterative_ensemble_smoother/ies.cpp [hunk 2]
```diff
@@ -425,7 +425,7 @@ MatrixXd makeX(const MatrixXd &A, const MatrixXd &Y0, const MatrixXd &R,
        and when the forward model is non-linear.
        Section 2.4.3
     */
-    if (A.rows() > 0 && A.cols() > 0) {
+    if (projection && (A.rows() > 0 && A.cols() > 0)) {
```
Comment (ManInFez) [inline src/iterative_ensemble_smoother/ies.cpp]: Should we even be here if A has either no columns or no rows?

File: src/iterative_ensemble_smoother/experimental.py (hunk 1 above)
Comment (dafeda) [reply src/iterative_ensemble_smoother/experimental.py]: Good point, thanks.

File: src/iterative_ensemble_smoother/ies.cpp (hunk 2 above)
Comment (dafeda) [reply src/iterative_ensemble_smoother/ies.cpp]: You are probably right, but I don't want to remove it in this PR.
Created an issue though:
https://github.com/equinor/iterative_ensemble_smoother/issues/44

**Review by ManInFez — APPROVED** [review]
LGTM

**Pull request merged**